src/charm.py

```diff
@@ -702,11 +702,32 @@ def _generate_prometheus_config(self) -> bool:
         scrape_jobs = self.metrics_consumer.jobs()
         for job in scrape_jobs:
             job["honor_labels"] = True
-            if (tls_config := job.get("tls_config")) and (ca_file := tls_config.get("ca_file")):
-                # Cert is transferred over relation data and needs to be written to a file on disk.
-                cert_filename = f"/etc/prometheus/{job['job_name']}.crt"
-                certs[cert_filename] = ca_file
-                job["tls_config"]["ca_file"] = cert_filename
+            if tls_config := job.get("tls_config"):
+                # Certs are transferred over relation data and need to be written to files on disk.
+                # CA certificate to validate the server certificate with.
+                if ca_file := tls_config.get("ca_file"):
+                    filename = f"{PROMETHEUS_DIR}/{job['job_name']}.crt"
+                    certs[filename] = ca_file
+                    job["tls_config"]["ca_file"] = filename
+                # Certificate and key files for client cert authentication to the server.
+                if (cert_file := tls_config.get("cert_file")) and (
+                    key_file := tls_config.get("key_file")
+                ):
+                    filename = f"{PROMETHEUS_DIR}/client-{job['job_name']}.crt"
+                    certs[filename] = cert_file
+                    job["tls_config"]["cert_file"] = filename
+                    filename = f"{PROMETHEUS_DIR}/client-{job['job_name']}.key"
```

Comment:
**suggestion (non-blocking)**: Name the files so they'd be sortable by jobname.

How about:
```diff
-{job['job_name']}.crt
-client-{job['job_name']}.crt
-client-{job['job_name']}.key
+{job['job_name']}-ca.crt
+{job['job_name']}-client.crt
+{job['job_name']}-client.key
```